contracts/facets/mmsa/MMSAFacet.sol

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.26;
+
+import {UserOperation} from "../../aa-4337/interfaces/UserOperation.sol";
+import {IAccount} from "../../aa-4337/interfaces/IAccount.sol";
+import {LibMMSAStorage, FallbackHandler} from "../../libraries/LibMMSAStorage.sol";
+import {LibSentinelList} from "../../libraries/LibSentinelList.sol";
+import {LibMode} from "../../libraries/LibMode.sol";
+import {LibVerification} from "../../libraries/LibVerification.sol";
+import {BarzStorage} from "../../libraries/LibAppStorage.sol";
+import {Executor} from "./utils/Executor.sol";
+import {ExecMode, CallType, ExecType, Execution, ValidationMode, ValidationId, ValidationType} from "./utils/Types.sol";
+import {VALIDATOR_MODULE_TYPE, EXECUTOR_MODULE_TYPE, FALLBACK_MODULE_TYPE, HOOK_MODULE_TYPE, CALLTYPE_SINGLE, CALLTYPE_STATIC, CALLTYPE_BATCH, EXECTYPE_DEFAULT, EXECTYPE_TRY} from "./utils/Constants.sol";
+import {ModuleManager} from "./utils/ModuleManager.sol";
+import {ValidationManager} from "./utils/ValidationManager.sol";
+import {IValidator} from "./interfaces/IValidator.sol";
+import {IModule} from "./interfaces/IModule.sol";
+import {IHook} from "./interfaces/IHook.sol";
+import {IMMSAFacet} from "./interfaces/IMMSAFacet.sol";
+
+/**
+ * @title MMSA(Minimal Modular Smart Account) Facet
+ * @dev Facet enabling Barz to be ERC-7579 compatible account
+ * @author David Yongjun Kim (@PowerStream3604)
+ */
+contract MMSAFacet is
+    IMMSAFacet,
+    IAccount,
+    ModuleManager,
+    ValidationManager,
+    Executor,
+    BarzStorage
+{
+    using LibSentinelList for LibSentinelList.SentinelList;
+    using LibMode for ExecMode;
+
+    uint256 private constant VALIDATION_FAILED = 1;
+    address private constant ENTRYPOINT =
```

Comment:
I think this will break 7579 compliance. as 7579 uses 0.7 Entrypoint and `PackedUserOperations`